include/geojson/Feature.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef GEOJSON_FEATURE_H
+#define GEOJSON_FEATURE_H
+
+#include "JSONGeometry.hpp"
+#include "JSONProperty.hpp"
+
+#include <memory>
+#include <ostream>
+#include <exception>
+#include <string>
+
+#include <boost/property_tree/ptree.hpp>
+
+namespace geojson {
+    enum class FeatureType {
+        None,
+        Point,
+        LineString,
+        Polygon,
+        MultiPoint,
+        MultiLineString,
+        MultiPolygon,
+        GeometryCollection
+    };
+
+    typedef std::map<std::string, JSONProperty> property_map;
+
+    class Feature {
+        public:
+            Feature(
+                JSONGeometry &new_geometry,
+                std::vector<double> new_bounding_box = std::vector<double>(),
+                property_map new_properties = property_map()
+            );
+
+            Feature(std::vector<JSONGeometry> new_geometry_collection,
+                std::vector<double> new_bounding_box = std::vector<double>(),
+                property_map new_properties = property_map()
+            );
+
+            Feature(boost::property_tree::ptree &tree);
+
+            Feature(const Feature &feature);
```

Comment:
If we are going to have a copy constructor, I would advocate for a swap function as well.  Copy Swap is a nice pattern for implementing various semantics, and might be a good idea to put in as many places as possible.  If performance issues arise we can profile and tackle them on an individual basis.